File: thworld.js
const puppeteer = require("puppeteer");
var log4js = require("log4js");
var logger = log4js.getLogger();
logger.level = "debug";

(async () => {
    const browser = await puppeteer.launch();
    const page = await browser.newPage({
        slowMo: 100
    });

    logger.info("开始签到");

    await page.goto("http://www.thworld.net/auth/login");

    await page.type("#email", "email");
    await page.type("#passwd", "passwd");

    logger.info("登录...");
    await page.click("#login");

    // 等待页面登录到首页，可以看到'签到获取流量'
    await page.waitFor(6000);

    logger.info("等待验证条加载...");

    try {
        // let iframe;

        // for (let child of page.mainFrame().childFrames()) {
        //     if (child.url().includes("https://authedmine.com")) {
        //         iframe = child;
        //         break;
        //     }
        // }
        // const verifyMe = await iframe.$("#verify-me");
        // console.log(await verifyMe.click());
        // dumpFrameTree(page.mainFrame(), "");
        // await browser.close();

        // function dumpFrameTree(frame, indent) {
        //     console.log(indent + frame.url());
        //     for (let child of frame.childFrames()) dumpFrameTree(child, indent + "  ");
        // }
        const iframe = page.frames().find(f => {
            return f.url().includes("https://authedmine.com");
        });

        console.log("iframe", iframe);
        // const verifyMe = await iframe.$("#verify-me");
        //         console.log(await verifyMe.text());
        // if (!iframe) {
        //     logger.info("已经签到");
        //     await browser.close();
        // }

        // logger.info("点击验证条，等待20秒...");

        // await iframe.click("#verify-me");

        // await page.waitFor(20000);

        // logger.info("开始签到...");
        // await page.click("#checkin");
        // logger.info("签到成功");
    } catch (e) {
        logger.error("签到失败:", e);
    }

    await browser.close();
})();

File: thworld.py
from selenium import webdriver
import time
driver = webdriver.Chrome(executable_path='chromedriver')

auth_url = "https://thworld.net/auth/login"

driver.get(auth_url)
# 设置等待时间，等待页面加载之后再操作
time.sleep(3)

elem_user = driver.find_element_by_id('email')
elem_user.clear()
elem_user.send_keys('email')
elem_pwd = driver.find_element_by_id('passwd')
elem_pwd.clear()
elem_pwd.send_keys('passwd')
time.sleep(1)
driver.find_element_by_id("login").click()
print('login...sleep 10s')
time.sleep(10)

try:
    # use xpath to switch frame
    # driver.switch_to.frame(driver.find_element_by_xpath("//iframe[contains(@src,'https://authedmine.com')]"))
    # use index to switch frame
    driver.switch_to.frame(0)
    print('click verify me')
    driver.find_element_by_id('verify-me').click()

    print('sleep 30s to load verify bar')
    time.sleep(30)
    driver.switch_to.default_content()

    time.sleep(3)

    print(driver.find_element_by_id('checkin-btn').text)
    driver.find_element_by_id('checkin').click()

    print('sleep 5s for checkin')
    time.sleep(5)
    driver.close()
except Exception as e:
    driver.close()
    print('checked in')